strawberry/printer.py

```diff
@@ -34,21 +40,55 @@
     from strawberry.schema import BaseSchema
 
 
-def print_schema_directive_params(params: Dict) -> str:
+def _serialize_dataclass(value: Any) -> Any:
+    if dataclasses.is_dataclass(value):
+        return dataclasses.asdict(value)
+    if isinstance(value, Iterable):
+        return [_serialize_dataclass(v) for v in value]
+    if isinstance(value, Mapping):
+        return {k: _serialize_dataclass(v) for k, v in value.items()}
+
+    return value
+
+
+def print_schema_directive_params(
+    directive: GraphQLDirective, values: Dict[str, Any]
+) -> str:
+    params = []
+    for name, arg in directive.args.items():
+        value = values.get(name, arg.default_value)
+        if is_unset(arg):
+            value = None
+        else:
+            ast = ast_from_value(value, arg.type)
+            value = ast and f"{name}: {print_ast(ast)}"
+
+        if value:
+            params.append(value)
+
     if not params:
         return ""
 
-    return "(" + ", ".join(f'{name}: "{value}"' for name, value in params.items()) + ")"
+    return "(" + ", ".join(params) + ")"
 
 
 def print_schema_directive(
     directive: StrawberrySchemaDirective, schema: BaseSchema
 ) -> str:
-    params = directive.instance.__dict__ if directive.instance else {}
+    from strawberry.schema.schema import Schema
+
+    if isinstance(schema, Schema):
+        schema_converter = schema.schema_converter
+    else:
+        schema_converter = GraphQLCoreConverter(schema.config, {})
```

Comment:
why do we need this check?

Reply:
Because `BaseSchema` doesn't have a `.schema_converter` in it, and those printers are expecting a `BaseSchema` instance.

Maybe `BaseSchema` should add a `schema_converter` to its protocol?

Reply:
yes, let's do that for now, I think we should revisit Schema and BaseSchema, as BaseSchema doesn't really provide any useful thing at the moment, we are still far away from having a full abstraction on GraphQL core :)

Reply:
Done!